packages/shared_preferences/shared_preferences_tool/lib/src/shared_preferences_tool_eval.dart

```diff
@@ -0,0 +1,340 @@
+    return _legacyEval.legacyPrefsGetInstance(
+      "get('$key')",
+      _isWeb,
+      _valueDisposable,
+    );
+  }
+}
+
+extension on EvalOnDartLibrary {
+  Future<InstanceRef?> prefsEval(
+    String method,
+    bool isWeb,
+    Disposable? isAlive,
+  ) async {
+    return evalFuture(
+      'SharedPreferencesAsync().$method',
+      isWeb,
+      isAlive,
+    );
+  }
+
+  Future<Instance> prefsGetInstance(
+    String method,
+    bool isWeb,
+    Disposable? isAlive,
+  ) async {
+    return safeGetInstance(
+      (await prefsEval(method, isWeb, isAlive))!,
+      isAlive,
+    );
+  }
+
+  /// This only works on non-web platforms due to the `asyncEval` call.
+  /// This is ok, this won't ever be called on web platforms.
+  Future<InstanceRef?> legacyPrefsEval(
+    String method,
+    bool isWeb,
+    Disposable? isAlive,
+  ) {
+    return evalFuture(
+      'SharedPreferences.getInstance().then((prefs) => prefs.$method)',
+      isWeb,
+      isAlive,
+    );
+  }
+
+  Future<Instance> legacyPrefsGetInstance(
+    String method,
+    bool isWeb,
+    Disposable? isAlive,
+  ) async {
+    return safeGetInstance(
+      (await legacyPrefsEval(method, isWeb, isAlive))!,
+      isAlive,
+    );
+  }
+
+  /// Evaluates the given [expression].
+  ///
+  /// Returns the [InstanceRef] of the result.
+  /// The [isAlive] parameter is used to dispose the evaluation if the
+  /// caller is disposed.
+  ///
+  /// This method is actually a workaround for the asyncEval method, which is
+  /// not working for web targets, check this issue https://github.com/flutter/devtools/issues/7766.
+  ///
+  /// It does the normal asyncEval on platforms other than web.
+  Future<InstanceRef?> evalFuture(
+    String expression,
+    bool isWeb,
+    Disposable? isAlive,
+  ) async {
+    // If not running on a web target, returns the normal async eval.
+    if (!isWeb) {
+      return asyncEval(
+        'await $expression',
+        isAlive: isAlive,
+      );
+    }
+
+    // Create a empty list in memory to hold the future value instance.
+    // It could've been anything that can handle values passed by reference.
+    final InstanceRef valueHolderRef = await safeEval(
+      '[]',
+      isAlive: isAlive,
+    );
+
+    // Add the future value instance to the list once the future completes
+    await safeEval(
+      '$expression.then(valueHolder.add);',
+      isAlive: isAlive,
+      scope: <String, String>{
+        'valueHolder': valueHolderRef.id!,
+      },
+    );
+
+    // The maximum number of retries to get the future value instance.
+    // Means a max of 1 second of waiting.
+    const int maxRetries = 20;
+    int retryCount = 0;
+
+    // Wait until the shared preferences instance is added to the list.
+    while (true) {
+      retryCount++;
+      // A break condition to avoid infinite loop.
+      if (retryCount > maxRetries) {
+        throw StateError('Failed to get future value instance.');
+      }
```

Comment:
Could this be `while retryCount <= maxRetries` and have the throw after the block?

I'm not sure if it's actually better, but seeing while(true) always worries me.

Reply:
yes! Definitely be better. Gonna change it.